ansible_ai_connect/ai/api/model_pipelines/llamastack/pipelines.py

```diff
@@ -146,27 +172,18 @@ def self_test(self) -> HealthCheckSummary:
                 MODEL_MESH_HEALTH_CHECK_MODELS: "ok",
             }
         )
-        try:
-            client = LlamaStackClient(base_url=self.config.inference_url)
-            response = client.providers.retrieve(LLAMA_STACK_PROVIDER_ID)
-            health_status = response.health.get("status")
-            if health_status != "OK":
-                reason = f"Provider {LLAMA_STACK_PROVIDER_ID} health status: {health_status}"
-                summary.add_exception(
-                    MODEL_MESH_HEALTH_CHECK_MODELS,
-                    HealthCheckSummaryException(ServiceUnavailable(reason)),
-                )
-        except BadRequestError as e:
-            logger.exception(str(e))
+        if not self.index_health():
+            reason = f"Provider {LLAMA_STACK_DB_PROVIDER} health status: Not ready"
             summary.add_exception(
                 MODEL_MESH_HEALTH_CHECK_MODELS,
```

Comment:
This should be `MODEL_MESH_HEALTH_CHECK_INDEX` (if you follow my proposal above).